### PR TITLE
Added classes necessary for handle OneVizion notification queue. Refactored Auth Token support.

### PR DESCRIPTION
**Pull request by Anastasiyaaa**

### CODE REVIEW

**Comment by Anastasiyaaa** [comment]
[Depl-168644](https://trackor.onevizion.com/form/ConfigAppForm.do?id=100099104479&ttid=10009161) Abstract class to handle OneVizion notification queue
In this issue added classes for handling the notification queue. This is necessary in order not to duplicate the code in the integrations associated with sending notifications (sms-notification, slack-notification, then, perhaps, others will appear)

[API-170071](https://trackor.onevizion.com/form/ConfigAppForm.do?id=100099504550&ttid=10009161) Refactoring Auth Token support in onevizion library
The code related to Auth Token support was refactored, including the name of the ovToken parameter was changed to isTokenAuth, because of this, an exception may occur in integrations where this parameter is used (in the OV Integrations organization it is used only in the ov-monitoring integration)

**Comment by KeyMasterOfGozer** [comment]
Why does the class name have "API" in it.

class NotifQueueApi:

Should it not be 

class NotifQueue:

File: onevizion.py [hunk 1]
```diff
@@ -1569,6 +1572,281 @@ def sendmail(self):
...
+
+
+    def getNotifQueue(self):
+        URL = "https://{URL}/api/internal/notif/queue?service_id={ServiceID}".format(URL=self._URL, ServiceID=self._serviceId)
+        OVCall = curl('GET', URL, headers=self._headers, auth=self._auth)
+        if len(OVCall.errors) > 0:
+            raise Exception(OVCall.errors)
+        return OVCall.jsonData
+
+    def updateNotifQueueRecStatus(self, notifQueueRecId, status):
```
Comment (glebowadim) [inline onevizion.py]: Let's rename this method to updateNotifQueueRecStatusById

File: onevizion.py [hunk 2]
```diff
@@ -1569,6 +1572,281 @@ def sendmail(self):
...
+        if len(OVCall.errors) > 0:
+            raise Exception(OVCall.errors)
+
+    def addNewAttempt(self, notifQueueRecId, errorMessage):
+        URL = "https://{URL}/api/internal/notif/queue/{notifQueueRecId}/attempts?error_code={errorMessage}".format(URL=self._URL, notifQueueRecId=notifQueueRecId, errorMessage=errorMessage)
+        OVCall = curl('POST', URL, headers=self._headers, auth=self._auth)
+        if len(OVCall.errors) > 0:
+            raise Exception(OVCall.errors)
+
+    def updateNotifQueueRecStatusByObject(self, notifQueueRec):
```
Comment (glebowadim) [inline onevizion.py]: Rename updateNotifQueueRecStatusByObject method to updateNotifQueueRecStatus

File: onevizion.py [hunk 3]
```diff
@@ -1569,6 +1572,281 @@ def sendmail(self):
...
 
 
+class NotificationService(ABC):
+    """Wrapper for getting records from the notification queue and sending them somewhere.
+        It is an abstract class whose 'sendNotification' method you must implement.
+
+	Attributes:
+        serviceId: ID of the Notification Service
+        processId: the system processId
+		URL: A string representing the website's main URL for instance "trackor.onevizion.com".
```
Comment (glebowadim) [inline onevizion.py:1582]: Please check and correct this description

**Review by glebowadim — CHANGES_REQUESTED** [review]

File: onevizion.py (hunk 1 above)
Comment (Anastasiyaaa) [reply onevizion.py]: Done

File: onevizion.py (hunk 2 above)
Comment (Anastasiyaaa) [reply onevizion.py]: Done

**Comment by Anastasiyaaa** [comment]
Code corrected, @KeyMasterOfGozer , please take another look.

**Pull request merged**